subsys/bluetooth/audio/has.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+
+#include <device.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/audio/audio.h>
+#include <bluetooth/audio/capabilities.h>
+
+#include "has_internal.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HAS)
+#define LOG_MODULE_NAME bt_has
+#include "common/log.h"
+
+static struct bt_has has;
+
+BUILD_ASSERT(!(CONFIG_BT_HAS_HEARING_AID_TYPE > 2));
```

Comment:
Given that we actually have the range `range 0 2` in the Kconfig option, I don't think this is necessary at all (nor is a mask necessary as @asbjornsabo suggested). 

We could perhaps add a reference to where the values 0, 1 and 2 comes from (I assume the HAS spec?) in the `help` section of the Kconfig so it's less likely to be changed accidentally.

Reply:
A reference is good.
HAS spec, table 3.2

Reply:
I could add a comment as CONFIG_BT_HAS_HEARING_AID_TYPE is hidden option